packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/serialization_descriptors.py

```diff
@@ -0,0 +1,107 @@
+"""Serialization Descriptors."""
+
+from json import loads
+from typing import List
+
+from google.protobuf import descriptor_pb2, descriptor_pool
+from google.protobuf.descriptor_pb2 import DescriptorProto, FieldDescriptorProto
+
+from ni_measurement_plugin_sdk_service._annotations import ENUM_VALUES_KEY
+from ni_measurement_plugin_sdk_service._internal.parameter._get_type import (
+    _TYPE_FIELD_MAPPING,
+)
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import (
+    ParameterMetadata,
+)
+
+
+def _create_enum_type_class(
+    file_descriptor: descriptor_pb2.FileDescriptorProto,
+    parameter_metadata: ParameterMetadata,
+    field_descriptor: FieldDescriptorProto,
+) -> None:
+    """Implement a enum class in 'file_descriptor'."""
+    enum_dict = loads(parameter_metadata.annotations[ENUM_VALUES_KEY])
+    enum_type_name = _get_enum_type(parameter_metadata).__name__
+
+    # if enum is a protobuf then enum_type_name is 1st letter of each enum name
+    # e.g. {"NONE": 0, "RED": 1, "GREEN": 2} -> NRG
+    if enum_type_name == "int" or enum_type_name == "NoneType":
```

Comment:
Comparing types by name seems kind of hacky. What if you did `enum_type = _get_enum_type(parameter_metadata)` and `if enum_type is int or enum_type is NoneType:`?